components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -486,88 +486,361 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
+     * @param offset             Starting index of the count
+     * @param limit              Counting value
+     * @param sortBy             SortBy
+     * @param sortOrder          Sorting order
+     * @param domainName         Domain to run the filter
+     * @return User list with detailed attributes
+     * @throws CharonException Error while filtering
+     */
+    private List<Object> getFilteredUsersBySingleAttribute(ExpressionNode node, Map<String, Boolean> requiredAttributes,
+            int offset, int limit, String sortBy, String sortOrder, String domainName) throws CharonException {
 
-                if (!SCIMConstants.UserSchemaConstants.GROUP_URI.equals(attributeName)) {
-                    //get the user name of the user with this id
-                    userNames = getUserNames(attributeName, filterOperation, attributeValue);
+        String[] userNames;
+
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Listing users by filter: %s %s %s", node.getAttributeValue(), node.getOperation(),
+                    node.getValue()));
+        }
+        try {
+            // Apply filter enhancements for username and update expression node value.
+            applyFilterEnhancementsForUsername(node);
+
+            if ((carbonUM instanceof PaginatedUserStoreManager) && isPaginatedUserStoreAvailable()) {
+
+                // If the limit is not specified, list all the users.
+                if (limit <= 0) {
+                    userNames = getAllFilteredUsers(node, limit, offset, domainName);
                 } else {
-                    if (filterOperation.equalsIgnoreCase(SCIMCommonConstants.EQ)) {
-                        userNames = carbonUM.getUserListOfRole(attributeValue);
-                    } else if (carbonUM instanceof AbstractUserStoreManager) {
-                        String[] roleNames = getRoleNames(filterOperation, attributeValue);
-                        userNames = getUserListOfRoles(roleNames);
-                    } else {
-                        throw new NotImplementedException(String.format("Filter operator %s is not supported " +
-                                "by the user store.", filterOperation));
-                    }
+                    // Extract he domain name if the domain name is embedded in the filter attribute value.
+                    domainName = resolveDomainNameInAttributeValue(domainName, node);
+
+                    // Get filtered user names.
+                    userNames = getFilteredUsers(node, offset, limit, sortBy, sortOrder, domainName);
                 }
-            } catch (UserStoreException e) {
-                throw new CharonException(String.format("Error in filtering users by attribute name: %s, " +
-                                "attribute value: %s and filter operation %s", attributeName, attributeValue,
-                        filterOperation), e);
+            } else
```

Comment:
Cover the else block with `{}`

Reply:
This is changed in the current master branch